source/fundamentals/connection.txt

```diff
@@ -0,0 +1,510 @@
+
+This guide shows you how to connect to a MongoDB instance or replica set
+deployment using the driver. For information about authenticating
+with a MongoDB instance, see :ref:`Authentication <authentication>`.
+
+.. _mongoclient:
+
+MongoClient
+-----------
+
+Use the ``MongoClients.create()`` method to construct a
+``MongoClient``. As each ``MongoClient`` represents a threadsafe *pool* of
+connections to the database, most applications only require a single
+instance of a ``MongoClient``, even across multiple threads. All resource
+usage limits, such as max connections, apply to individual
+``MongoClient`` instances.
+
+.. note::
+
+   Always call ``MongoClient.close()`` to clean up resources when an
+   instance is no longer needed.
+
+.. _connection-uri:
+
+Connection URI
+--------------
+
+The **connection URI** is the set of instructions that the driver uses to
+connect to a MongoDB deployment. It instructs the driver on where and how
+it should connect to MongoDB as well as how it should behave while
+connected. The following example shows each part of the connection URI:
+
+.. figure:: /includes/figures/connection_string_parts.png
+
+In this example, for the protocol, we use ``mongodb+srv`` which specifies the
+:manual:`DNS Seedlist Connection Format
+</reference/connection-string/#dns-seedlist-connection-format>`. This indicates
+that the hostname following it corresponds to the DNS SRV record of your
+MongoDB instance or deployment. If your instance or deployment does not have a
+DNS SRV record, use ``mongodb`` to specify the :manual:`Standard Connection
+Format </reference/connection-string/#standard-connection-string-format>`.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, follow the
+   :atlas:`Atlas driver connection guide <driver-connection?jmp=docs_driver_java>`
+   to retrieve your connection string.
+
+The next part of the connection string contains your username and password
+if you are using password-based authentication. Replace the value of ``user``
+with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example below, we use ``sample-hostname``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string contains connection and authentication
+options as parameters. In the example below, we set two connection options:
+``retryWrites=true`` and ``w=majority``. For more information on connection
+options, skip to the :ref:`connection options <connection-options>` section.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.java
+   :language: java
+
+.. _jndi:
```

Comment:
I think moving JNDI to a separate section as the current Java reference docs are structured could be helpful in separating concerns. I also think connecting to a replica set should be directly after Connection URI since it consists of passing different values to the URI that was just introduced. What do you think?

Reply:
sounds good